### PR TITLE
Widgets - Validate

### PR DESCRIPTION
**Pull request by pannga**
Fixed some problems with Validate widget + fixed trackingId error

### CODE REVIEW

**Review by darekg11 — CHANGES_REQUESTED** [review]
Does not work at all.
Set it up without amount field and widget ends up sending: `&amount=NaN` and it always comes back with `400 - Bad request`

EDIT: God damnit, nevermind. My bad.

**Comment by pannga** [comment]
*tips fedora* thank you

**Comment by pannga** [comment]
I've simplified this app state as @darekg11 requested.

I've also removed this `withCredentials` attribute which I've added here by mistake -> I won't be adding this in the final version (I've talked about this with @pawelrychlik)

I still wait for PR #47 to be merged in order for this PR to pass linting.

**Comment by pannga** [comment]
@darekg11 I've merged those types and widget is ready for testing.

File: packages/react-widget/src/VoucherifyValidate.tsx [hunk 1]
```diff
@@ -1,13 +1,19 @@
 import * as React from 'react'
```
Comment (darekg11) [inline packages/react-widget/src/VoucherifyValidate.tsx:1]: Remove amount input, use gift card, get 400 response but all of the controls are in green color indicating like it was a success while in reality user got 400

File: packages/sdk/src/types/ClientSide.ts [hunk 2]
```diff
@@ -9,9 +9,12 @@ export interface ClientSideValidateParams {
 	code?: string
```
Comment (darekg11) [inline packages/sdk/src/types/ClientSide.ts:9]: Did you rebase your branch on top of master? I don't get why I see the same changes here which already have been applied in previous commit: https://github.com/voucherifyio/voucherify-js-sdk/commit/bf4e2ffdf018c4e6de9d256ea4a25e9b2d280cc6

File: packages/sdk/src/types/Validations.ts [hunk 3]
```diff
@@ -22,6 +22,9 @@ export interface ValidationsValidateVoucherParams {
 	gift?: {
```
Comment (darekg11) [inline packages/sdk/src/types/Validations.ts:22]: Did you rebase your branch on top of master? I don't get why I see the same changes here which already have been applied in previous commit: bf4e2ff

File: packages/sdk/src/types/ClientSide.ts [hunk 4]
```diff
@@ -121,6 +142,23 @@ export interface VoucherifyRedeemInputs {
 	voucherifyAmount: string
 	voucherifyTracking: string
 }
+
+export interface VoucherifyValidateInputsState {
```
Comment (darekg11) [inline packages/sdk/src/types/ClientSide.ts:146]: Let's please move those types which are strictly related to widget to some new file in `packages/react-widget` dir. It does not make sense for someone who only downaloaded SDK package to see those types.

**Review by darekg11 — CHANGES_REQUESTED** [review]
Small improvements required.

**Pull request closed without merging**